platform/posix/transport/src/openssl_posix.c

```diff
@@ -244,12 +244,15 @@ static OpensslStatus_t tlsHandshake( const ServerInfo_t * pServerInfo,
     int32_t sslStatus = -1, verifyPeerCertStatus = X509_V_OK;
 
     /* Validate the hostname against the server's certificate. */
-    sslStatus = SSL_set1_host( pOpensslParams->pSsl, pServerInfo->pHostName );
-
-    if( sslStatus != 1 )
+    if( pOpensslCredentials->disableHostnameCheck == 0U )
```

Comment:
I would prefer that we add a separate connect() function which takes a sockaddr_in rather than a hostname.

This would encourage the use of SNI whenever possible.

Reply:
But I think just checking for pOpensslCredentials->sniHostName != NULL should be sufficient here.